lib/config/folderconfiguration.go

```diff
@@ -206,6 +207,15 @@ func (f *FolderConfiguration) prepare(myID protocol.DeviceID, existingDevices ma
 	if f.MarkerName == "" {
 		f.MarkerName = DefaultMarkerName
 	}
+	if f.MarkerName == "." {
+		l.Warnf(`Setting the folder marker name to "." is not allowed, reverting to default "%v"`, DefaultMarkerName)
+		f.MarkerName = DefaultMarkerName
+	}
+	markerPath := filepath.Join(f.Path, f.MarkerName)
+	if s := strings.TrimPrefix(markerPath, f.Path); s == markerPath || s == "" {
+		l.Warnf(`The folder marker name "%v" is not located inside the folder path, reverting to default "%v"`, f.MarkerName, DefaultMarkerName)
```

Comment:
%q may catch some configuration errors that "%v" doesn't.

Reply:
I don't understand. `%q` escapes various characters, I just want to put quotes around the string that may contain spaces.

Reply:
I think greatroars point is that with %q it'll be obvious if the string contains for example a unicode nonbreaking space instead of a normal space, etc, which in some theoretical situation could be confusing if there was a file on disk with a normal space and we're debugging why Syncthing doesn't see it...